products/appengine/api.yaml

```diff
@@ -459,6 +459,800 @@ objects:
+      - !ruby/object:Api::Type::KeyValuePairs
+        name: 'betaSettings'
+        description: |
+          Metadata settings that are supplied to this version to enable beta runtime features.
+      - !ruby/object:Api::Type::Enum
+        name: 'servingStatus'
+        description: |
+          Current serving status of this version. Only the versions with a SERVING status create instances and can be billed.
+          Defaults to SERVING.
+        default_value: :SERVING
+        values:
+          - :SERVING
+          - :STOPPED
+      - !ruby/object:Api::Type::String
+        name: 'runtimeApiVersion'
+        description: |
+          The version of the API in the given runtime environment. 
+          Please see the app.yaml reference for valid values at https://cloud.google.com/appengine/docs/standard//config/appref
+      - !ruby/object:Api::Type::String
+        name: 'runtimeMainExecutablePath'
+        description: |
+          The path or name of the app's main executable.
+      - !ruby/object:Api::Type::Array
+        name: 'handlers'
+        description: |
+          An ordered list of URL-matching patterns that should be applied to incoming requests. 
+          The first matching URL handles the request and other request handlers are not attempted.
+        item_type: !ruby/object:Api::Type::NestedObject
+          properties:
+          - !ruby/object:Api::Type::String
+            name: 'urlRegex'
+            required: true
+            description: |
+              URL prefix. Uses regular expression syntax, which means regexp special characters must be escaped, but should not contain groupings. 
+              All URLs that begin with this prefix are handled by this handler, using the portion of the URL after the prefix as part of the file path.
+          - !ruby/object:Api::Type::Enum
+            name: 'securityLevel'
+            description: |
+              Security (HTTPS) enforcement for this URL.
+            values:
+              - :SECURE_UNSPECIFIED
+              - :SECURE_DEFAULT
+              - :SECURE_NEVER
+              - :SECURE_OPTIONAL
+              - :SECURE_ALWAYS
+          - !ruby/object:Api::Type::Enum
+            name: 'login'
+            description: |
+              Methods to restrict access to a URL based on login status.
+            values:
+              - :LOGIN_UNSPECIFIED
+              - :LOGIN_OPTIONAL
+              - :LOGIN_ADMIN
+              - :LOGIN_REQUIRED
+          - !ruby/object:Api::Type::Enum
+            name: 'authFailAction'
+            description: |
+              Actions to take when the user is not logged in.
+            values:
+              - :AUTH_FAIL_ACTION_UNSPECIFIED
+              - :AUTH_FAIL_ACTION_REDIRECT
+              - :AUTH_FAIL_ACTION_UNAUTHORIZED
+          - !ruby/object:Api::Type::Enum
+            name: 'redirectHttpResponseCode'
+            description: |
+              Redirect codes.
```

Comment:
Is this a user supplied value? If yes is there a better description of how it should be used?